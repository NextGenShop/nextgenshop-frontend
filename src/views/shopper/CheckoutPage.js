import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import StepForm from "../../components/StepForm";

const useStyles = makeStyles((theme) => ({
  appBar: {
    position: "relative",
    color: "primary",
  },
  layout: {
    width: "auto",
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
      width: 600,
      marginLeft: "auto",
      marginRight: "auto",
    },
  },
  paper: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
    padding: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
      marginTop: theme.spacing(6),
      marginBottom: theme.spacing(6),
      padding: theme.spacing(3),
    },
  },
  stepper: {
    padding: theme.spacing(3, 0, 5),
  },
  buttons: {
    display: "flex",
    justifyContent: "flex-end",
  },
  button: {
    marginTop: theme.spacing(3),
    marginLeft: theme.spacing(1),
  },
}));

export function Checkout() {
  const classes = useStyles();

  // TODO:
  // map through basket.items and render the product name, price, quantity etc.
  //console.log(basket); // REMOVE THIS (for testing purposes)

  return (
    <React.Fragment>
      <CssBaseline />
      <AppBar position="absolute" color="primary" className={classes.appBar}>
        <Toolbar>
          <Typography variant="h6" color="inherit" noWrap>
            <Link to="/login" style={{ textDecoration: "none", color: "#FFF" }}>
              NextGenShop
            </Link>
          </Typography>
        </Toolbar>
      </AppBar>
      <main className={classes.layout}>
        <Paper className={classes.paper}>
          <StepForm />
        </Paper>
      </main>
    </React.Fragment>
  );
}

const mapStateToProps = (state) => ({
  basket: state.basket,
});

export default connect(mapStateToProps, null)(Checkout);
