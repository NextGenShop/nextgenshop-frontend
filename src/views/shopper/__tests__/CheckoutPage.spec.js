import React from 'react';
import { shallow } from 'enzyme';
import { Checkout } from '../CheckoutPage';
import mockBasketItems from '../../../store/mock/MockBasketItems.json';

describe('CheckoutPage', () => {
  test('should correctly render', () => {
    const component = shallow(<Checkout />);
    expect(component).toMatchSnapshot();
  });
  test('should correctly render with props', () => {
    const mockBasket = { items: mockBasketItems, totalPrice: 0, shopper: 0 };
    const component = shallow(<Checkout basket={mockBasket} />);
    expect(component).toMatchSnapshot();
  });
});
