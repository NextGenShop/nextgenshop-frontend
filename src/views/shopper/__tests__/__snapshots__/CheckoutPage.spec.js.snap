// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage should correctly render 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-1"
    color="primary"
    position="absolute"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        <Link
          style={
            Object {
              "color": "#FFF",
              "textDecoration": "none",
            }
          }
          to="/login"
        >
          NextGenShop
        </Link>
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <main
    className="makeStyles-layout-2"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-3"
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        component="h1"
        variant="h4"
      >
        Checkout
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Stepper))
        activeStep={0}
        className="makeStyles-stepper-4"
      >
        <WithStyles(ForwardRef(Step))
          key="Shipping address"
        >
          <WithStyles(ForwardRef(StepLabel))>
            Shipping address
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
        <WithStyles(ForwardRef(Step))
          key="Payment details"
        >
          <WithStyles(ForwardRef(StepLabel))>
            Payment details
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
        <WithStyles(ForwardRef(Step))
          key="Review your order"
        >
          <WithStyles(ForwardRef(StepLabel))>
            Review your order
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
      </WithStyles(ForwardRef(Stepper))>
      <AddressForm />
      <div
        className="makeStyles-buttons-5"
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-button-6"
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          Next
        </WithStyles(ForwardRef(Button))>
      </div>
    </WithStyles(ForwardRef(Paper))>
  </main>
</Fragment>
`;

exports[`CheckoutPage should correctly render with props 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-1"
    color="primary"
    position="absolute"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        <Link
          style={
            Object {
              "color": "#FFF",
              "textDecoration": "none",
            }
          }
          to="/login"
        >
          NextGenShop
        </Link>
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <main
    className="makeStyles-layout-2"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-3"
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        component="h1"
        variant="h4"
      >
        Checkout
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Stepper))
        activeStep={0}
        className="makeStyles-stepper-4"
      >
        <WithStyles(ForwardRef(Step))
          key="Shipping address"
        >
          <WithStyles(ForwardRef(StepLabel))>
            Shipping address
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
        <WithStyles(ForwardRef(Step))
          key="Payment details"
        >
          <WithStyles(ForwardRef(StepLabel))>
            Payment details
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
        <WithStyles(ForwardRef(Step))
          key="Review your order"
        >
          <WithStyles(ForwardRef(StepLabel))>
            Review your order
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
      </WithStyles(ForwardRef(Stepper))>
      <AddressForm />
      <div
        className="makeStyles-buttons-5"
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-button-6"
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          Next
        </WithStyles(ForwardRef(Button))>
      </div>
    </WithStyles(ForwardRef(Paper))>
  </main>
</Fragment>
`;
